core/src/packet.rs

```diff
@@ -238,7 +242,7 @@ impl Packets {
                     }
                     trace!("got {} packets", npkts);
                     i += npkts;
-                    if npkts != NUM_RCVMMSGS || i >= 1024 {
```

Comment:
this check for NUM_RCVMMSGS is important, it's a signal that there's nothing there now, do you propose we poll for a full millisecond instead?

Reply:
Yes, if the buffer is not full. This early exit creates a lot of really small packet buffers which clog up the banking stage. Small batch performance is much worse than larger batches. I think it's better to take the latency hit early in the pipeline to allow them to clear faster later in the pipeline.